src/consensus/tx_check.h

```diff
@@ -5,6 +5,8 @@
 #ifndef BITCOIN_CONSENSUS_TX_CHECK_H
 #define BITCOIN_CONSENSUS_TX_CHECK_H
 
+#include <consensus/validation.h>
```

Comment:
You can still forward declare `TxValidationState` and `BlockValidationState` where needed and avoid the issues pointed by @ryanofsky?

Reply:
This causes a compilation error:

```console
In file included from consensus/tx_check.cpp:8:
./consensus/validation.h:115:7: error: typedef redefinition with different types ('ValidationState<TxValidationResult>' vs 'TxValidationState')
using TxValidationState = ValidationState<TxValidationResult>;
      ^
./consensus/tx_check.h:16:7: note: previous definition is here
class TxValidationState;
```

I'd have to forward declare the full template type and enum then use that in the function declaration.

```c++
enum class TxValidationResult;
template <typename Result>
class ValidationState;

bool CheckTransaction(const CTransaction& tx, ValidationState<TxValidationResult>& state);
```

Seems a bit verbose and would mismatch the function definition. FWIW, here are some [pros and cons](https://google.github.io/styleguide/cppguide.html#Forward_Declarations) of forward declaring that I'm aware of.